### PR TITLE
EFA summer cleanups

### PR DESCRIPTION
**Pull request by galpress**
A few cleanups to the EFA provider.
No functional changes in this series.

### CODE REVIEW

File: providers/efa/verbs.c [hunk 1]
```diff
@@ -1118,6 +1118,24 @@ static void efa_sq_advance_post_idx(struct efa_qp *qp)
...
+	udma_to_device_barrier();
+	efa_wq_ring_doorbell(&rq->wq, desc_idx);
+}
+
+static inline void efa_sq_ring_doorbell(struct efa_sq *sq, uint16_t desc_idx)
+{
+	mmio_flush_writes();
+	efa_wq_ring_doorbell(&sq->wq, desc_idx);
+}
+
```
Comment (rleon) [inline providers/efa/verbs.c:1145]: Let's do not add extra obfuscation. You added more lines than deleted and it is clear indication to me that extra functions are not needed.

Comment (galpress) [reply providers/efa/verbs.c:1145]: I don't think the primitive of ringing the doorbell should be hard coded more than once in different flows.
What do you think about dropping efa_wq_ring_doorbell and use the mmio_write32 directly in the RQ/SQ functions?

Comment (rleon) [reply providers/efa/verbs.c:1145]: If you can't leave without it, let's do it.

Comment (galpress) [reply providers/efa/verbs.c:1139]: Done

**Pull request merged**